src/io/read.rs

```diff
@@ -26,24 +38,89 @@ pub trait Read {
 
     /// Returns the number of bytes still available to read.
     fn remaining_data(&self) -> usize;
+}
+assert_obj_safe!(Read);
 
+/// Convenience functions for reading integers from a [`Read`].
+pub trait ReadInt: Read {
     /// Reads a little-endian integer.
     ///
     /// # Note
     /// Do not implement this function yourself. Callers are not required to
     /// call it in order to actually perform a read, so whether or not it is
     /// called is an implementation detail.
     #[inline]
-    fn read_le<I: LeInt>(&mut self) -> Result<I, io::Error>
-    where
-        Self: Sized,
-    {
+    fn read_le<I: LeInt>(&mut self) -> Result<I, io::Error> {
         I::read_from(self)
     }
 }
-assert_obj_safe!(Read);
 
-impl<R: Read + ?Sized> Read for &'_ mut R {
+impl<R: Read + ?Sized> ReadInt for R {}
+
+/// A [`Read`] that can zero-copy read data of a particular lifetime.
```

Comment:
I think the wording here is misleading. They way I read it, it sounds like every instance of `ReadZero` can perform zero-copy reads, which is not the case. `ReadZero` provides an interface that supports, but does not guarantee, zero-copy reads.

The function comment on `read_direct` has the same issue. When I see `Attempts to perform a "direct read" with the given layout`, I assume that if a direct read isn't possible the read will fail, not fall back to a copying read.